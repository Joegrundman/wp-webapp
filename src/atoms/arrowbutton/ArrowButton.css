.button {
  height: 2rem;
  width: 2rem;
  background-color: #2a2a2a;
  color: #cdcdcd;
  margin: 0 0.5rem;
  outline: none;
  border-radius: 3px;
}

.button:hover {
  color: #fafafa;
  border-color: #fafafa;
}

.text {
  font-size: 2rem;
  margin-top: -7px;
  margin-left: -1px;
}