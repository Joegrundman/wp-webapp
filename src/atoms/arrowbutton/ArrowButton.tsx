import * as React from 'react';
import locals from './ArrowButton.css';

export interface IArrowButtonProps {
  direction?: string,
  action: () => any
}

const ArrowButton: React.SFC<IArrowButtonProps> = ({
  action, 
  direction = 'left'
}: IArrowButtonProps): JSX.Element => (
  <button
    className={locals.button}
    onClick={action}>
    <div className={locals.text}>
      {direction === 'left' ? '<' : '>'}
    </div>
  </button>
)

export default ArrowButton