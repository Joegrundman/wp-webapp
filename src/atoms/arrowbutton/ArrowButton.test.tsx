import { shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import ArrowButton, { IArrowButtonProps } from './ArrowButton'

describe('<ArrowButton/>', () => {
  const defaultProps: IArrowButtonProps = {
    action: jest.fn(),
    direction: '',
  }

  it('should render as expected with default props', () => {
    const component: ShallowWrapper = shallow(<ArrowButton {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })

  it('should render as expected with direction right', () => {
    const props: IArrowButtonProps = {...defaultProps, direction: 'right'}
    const component: ShallowWrapper = shallow(<ArrowButton {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('should render as expected with direction left', () => {
    const props: IArrowButtonProps = {...defaultProps, direction: 'left'}
    const component: ShallowWrapper = shallow(<ArrowButton {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('should have an action that is triggered when clicked', () => {
    const component: ShallowWrapper = shallow(<ArrowButton {...defaultProps} />)
    component.find('button').first().simulate('click')
    expect(defaultProps.action).toHaveBeenCalled()
  })
})
