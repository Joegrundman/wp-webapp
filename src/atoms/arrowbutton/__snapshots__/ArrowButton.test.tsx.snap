// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowButton/> should render as expected with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArrowButton
    action={[MockFunction]}
    direction=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
      >
        &gt;
      </div>,
      "className": undefined,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": ">",
        "className": undefined,
      },
      "ref": null,
      "rendered": ">",
      "type": "div",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
        >
          &gt;
        </div>,
        "className": undefined,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ">",
          "className": undefined,
        },
        "ref": null,
        "rendered": ">",
        "type": "div",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ArrowButton/> should render as expected with direction left 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArrowButton
    action={[MockFunction]}
    direction="left"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
      >
        &lt;
      </div>,
      "className": undefined,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "<",
        "className": undefined,
      },
      "ref": null,
      "rendered": "<",
      "type": "div",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
        >
          &lt;
        </div>,
        "className": undefined,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<",
          "className": undefined,
        },
        "ref": null,
        "rendered": "<",
        "type": "div",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<ArrowButton/> should render as expected with direction right 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArrowButton
    action={[MockFunction]}
    direction="right"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className={undefined}
      >
        &gt;
      </div>,
      "className": undefined,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": ">",
        "className": undefined,
      },
      "ref": null,
      "rendered": ">",
      "type": "div",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className={undefined}
        >
          &gt;
        </div>,
        "className": undefined,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ">",
          "className": undefined,
        },
        "ref": null,
        "rendered": ">",
        "type": "div",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
